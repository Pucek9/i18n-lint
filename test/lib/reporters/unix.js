/**
 * hslint default reporter tests
 *
 * @TODO make stdout hooking DRY
 *
 * Copyright (c) 2014 James Warwood
 * Licensed under the MIT license.
 */
/* global describe, it */
'use strict';

// Core imports
var fs = require('fs');

// npm imports
var expect = require('chai').expect;
var hooker = require('hooker');

// Local imports
var errors = require('../../fixtures/errors/1.js')();
var reporter = require('../../../lib/reporters/unix.js');

describe('hslint reporters unix', function() {
  it('should format the output correctly', function() {

    // Setup
    var actual = '';

    hooker.hook(process.stdout, 'write', {
      pre: function(out) {
        actual += out;

        return hooker.preempt();
      }
    });

    var expected = fs.readFileSync(__dirname +
        '/../../expected/reporters/unix.txt').toString();

    // Execute
    reporter(errors);
    hooker.unhook(process.stdout, 'write');

    // Assert
    expect(actual).to.equal(expected);
  });

  it('should not output anything if there are no errors', function() {
    var actual = '';

    hooker.hook(process.stdout, 'write', {
      pre: function(out) {
        actual += out;

        return hooker.preempt();
      }
    });

    // Execute
    reporter([]);

    hooker.unhook(process.stdout, 'write');

  });
});
