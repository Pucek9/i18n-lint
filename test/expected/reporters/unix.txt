somefile.html:5:5 Hardcoded <p> tag
somefile.html:7:19 Hardcoded 'alt' attribute
